File: app/src/main/java/com/paradigmadigital/dagger/ui/another/AnotherActivity.java
package com.paradigmadigital.dagger.ui.another;

import com.paradigmadigital.dagger.R;
import com.paradigmadigital.dagger.ui.BaseActivity;

import android.os.Bundle;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class AnotherActivity extends BaseActivity {

    @Inject
    AnotherPresenter presenter;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_another);

        presenter.initialize();
    }
}

File: app/src/main/java/eu/sesma/dagger/another_activity/AnotherActivity.kt
package eu.sesma.dagger.another_activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import eu.sesma.dagger.R
import eu.sesma.dagger.di.DaggerActivityComponent
import eu.sesma.dagger.platform.ActivityModule
import eu.sesma.dagger.platform.AndroidApplication
import javax.inject.Inject

class AnotherActivity : AppCompatActivity() {

    @Inject
    lateinit var presenter: AnotherPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_another)
        DaggerActivityComponent.builder()
                .applicationComponent((application as AndroidApplication).applicationComponent)
                .activityModule(ActivityModule(this))
                .build().inject(this)

        presenter.initialize()
    }
}
